### PR TITLE
Additionally handle logstart and logfinish hooks

### PR DESCRIPTION
**Pull request by s0undt3ch**
Thanks for submitting a PR, your contribution is really appreciated!

Here's a quick checklist that should be present in PRs:

- [x] Add a new news fragment into the changelog folder
  * name it `$issue_id.$type` for example (588.bugfix)
  * if you don't have an issue_id change it to the pr id after creating the pr
  * ensure type is one of `removal`, `feature`, `bugfix`, `vendor`, `doc` or `trivial`
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."
- [x] Target: for `bugfix`, `vendor`, `doc` or `trivial` fixes, target `master`; for removals or features target `features`;
- [x] Make sure to include reasonable tests for your change if necessary

Unless your change is a trivial or a documentation fix (e.g.,  a typo or reword of a small section) please:

- [x] Add yourself to `AUTHORS`, in alphabetical order;

### CODE REVIEW

File: _pytest/logging.py [hunk 1]
```diff
@@ -371,18 +371,20 @@ def _runtest_for(self, item, when):
                            formatter=self.formatter, level=self.log_level) as log_handler:
             if self.log_cli_handler:
                 self.log_cli_handler.set_when(when)
```
Comment (thisch) [inline _pytest/logging.py]: I would add the following code here and remove your `if item` checks:

```py
if item is None:
    yield
    return
```

**Comment by thisch** [comment]
Thx for working on this!

File: _pytest/logging.py (hunk 1 above)
Comment (s0undt3ch) [reply _pytest/logging.py]: Good point.

**Comment by s0undt3ch** [comment]
@nicoddemus sorry, I force pushed the changes asked by @thisch and lost your change log changes(forgot to pull first).

**Comment by nicoddemus** [comment]
> I force pushed the changes asked by @thisch and lost your change log changes(forgot to pull first).

No worries, just pushed my change again. 👍

**Comment by s0undt3ch** [comment]
Thanks!

File: _pytest/logging.py [hunk 2]
```diff
@@ -371,10 +371,17 @@ def _runtest_for(self, item, when):
...
+
+            if item is None:
+                yield  # run the test
+                return
+
             if not hasattr(item, 'catch_log_handlers'):
                 item.catch_log_handlers = {}
+
```
Comment (thisch) [inline _pytest/logging.py]: This newline and the next one are not really needed. Can you remove them please?

Comment (s0undt3ch) [reply _pytest/logging.py]: Done

Comment (thisch) [reply _pytest/logging.py]: Thx

**Pull request merged**